src/main/java/seedu/address/ui/CommissionDetailsPane.java

```diff
@@ -0,0 +1,118 @@
+
+import java.util.Comparator;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.ObservableObject;
+import seedu.address.model.commission.Commission;
+import seedu.address.model.commission.Description;
+import seedu.address.model.iteration.Iteration;
+
+/**
+ * An UI component that displays information about a {@code Commission}.
+ */
+public class CommissionDetailsPane extends UiPart<Region> {
+
+    private static final String FXML = "CommissionDetailsPane.fxml";
+
+    private final Commission commission;
+
+    @FXML
+    private HBox detailsPane;
+    @FXML
+    private Label title;
+    @FXML
+    private Label description;
+    @FXML
+    private Label fee;
+    @FXML
+    private Label deadline;
+    @FXML
+    private Label completionStatus;
+    @FXML
+    private Label customerName;
+    @FXML
+    private FlowPane tags;
+    @FXML
+    private ListView<Iteration> iterationListView;
+
+    /**
+     * Creates a {@code CommissionDetailsPane} with the given {@code Commission} and index to display.
+     */
+    public CommissionDetailsPane(ObservableObject<Commission> commission) {
+        super(FXML);
+        this.commission = commission.getValue();
+        updateUI(this.commission);
+        commission.addListener((observable, oldValue, newValue) -> {
+            updateUI(newValue);
+        });
+        iterationListView.setItems(commission.getValue().getIterationList());
+        iterationListView.setCellFactory(listView -> new IterationListViewCell());
+    }
+
+    private void updateUI(Commission commission) {
+        if (commission == null) {
+            title.setText("No commission selected");
+            description.setText("");
+            fee.setText("");
+            deadline.setText("");
+            completionStatus.setText("");
+            customerName.setText("");
+            tags.getChildren().clear();
+        } else {
+            title.setText(commission.getTitle().title);
+            description.setText(commission.getDescription().orElseGet(() -> Description.NO_DESCRIPTION).description);
+            fee.setText(commission.getFee().toString());
+            deadline.setText(commission.getDeadline().toString());
+            completionStatus.setText(commission.getCompletionStatus().toString());
+            customerName.setText(commission.getCustomer().getName().fullName);
+            tags.getChildren().clear();
+            commission.getTags().stream()
+                    .sorted(Comparator.comparing(tag -> tag.tagName))
+                    .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+            iterationListView.setItems(commission.getIterationList());
+            iterationListView.setCellFactory(listView -> new IterationListViewCell());
```

Comment:
Just wondering does the UI work now, remembered when you discuss with Xing you it was the ListView problem, now it's used?

Reply:
Oh, previously I was wondering how to make the GUI responsive to an `ObservableList` because you can't really `addListener` to an `ObservableList` like in `ObservableObject`.

My initial implementation was simply iterating through the `UniqueIterationList` and creating an `IterationListItem` for each iteration in the `UniqueIterationList`. So, when the `UniqueIterationList` is updated (by adding a new Iteration or deleting an Iteration), the GUI doesn't know because no listeners are attached to the `ObservableList`.

Xing Yu's suggestion was to use a `ListView` instead (which is how the `CommissionListPanel` and `CustomerListPanel` is being rendered). Not very sure how it works, but through some JavaFx magic it basically takes in an `ObservableList` and acts as a Listener it and it works 😄